arangod/Replication/TailingSyncer.cpp

```diff
@@ -369,8 +373,7 @@ Result TailingSyncer::processDBMarker(TRI_replication_operation_e type,
 
       auto system = sysDbFeature.use();
       TRI_ASSERT(system.get());
-      // delete from cache by id and name
-      _state.vocbases.erase(std::to_string(vocbase->id()));
```

Comment:
Why is it removed?

Reply:
Accessing databases by their id instead of their name was some very old (pre-3.3) behavior in old replication code.
It should not be needed anymore.
Furthermore, I added an assertion that makes sure no databases with numeric names get inserted into `_state.vocbases` here:
https://github.com/arangodb/arangodb/pull/18405/files#diff-c1e7fa959d484984ca6af502be6f161c7ee02fe37ac6f9a47072256851d11fe7R556

Reply:
👍